internal/graph/check.go

```diff
@@ -422,12 +437,23 @@ func exclusion(ctx context.Context, concurrencyLimit uint32, handlers ...CheckHa
 		}
 	}
 
-	if errs.ErrorOrNil() != nil {
-		return response, errs
+	if (baseErr != nil && !errors.Is(baseErr, ErrCycleDetected)) &&
+		(subErr != nil && !errors.Is(subErr, ErrCycleDetected)) {
+		return &ResolveCheckResponse{
+			Allowed: false,
+			ResolutionMetadata: &ResolutionMetadata{
+				DatastoreQueryCount: 0,
+			},
+		}, errors.Join(baseErr, subErr)
+	}
+
+	allowed := true
+	if errors.Is(baseErr, ErrCycleDetected) {
+		allowed = false
```

Comment:
IMO it's hard to reason about this. Can we not just handle the `ErrCycleDetected` in the channel handlers above?

Reply:
Changed to simplify.